src/customFilters.test.js

```diff
@@ -24,4 +28,202 @@ multiline`);
   const expected = 'This is multiline';
   
   is(value, expected);
+});
+
+test('.getPayloadExamples() should return epmty examples', t => {
+  const result = getPayloadExamples(
+    new Message({
+      examples: [
+        {
+          name: exampleName,
+          summary: exampleSummary,
+          headers: { foo: 'bar' },
+        },
+        {
+          name: exampleName,
+          summary: exampleSummary,
+          headers: { bar: 'foo' },
+        },
+      ],
+    }),
+  );
+  t.is(result, undefined);
+});
+
+test('.getPayloadExamples() should return payload examples', t => {
+  const result = getPayloadExamples(
+    new Message({
+      examples: [
+        {
+          name: exampleName,
+          summary: exampleSummary,
+          payload: { foo: 'bar' },
+        },
+        {
+          name: exampleName,
+          summary: exampleSummary,
+          payload: { bar: 'foo' },
+        },
+      ],
+    }),
+  );
+  t.deepEqual(result, [
+    {
+      name: exampleName,
+      summary: exampleSummary,
+      example: { foo: 'bar' },
+    },
+    {
+      name: exampleName,
+      summary: exampleSummary,
+      example: { bar: 'foo' },
+    },
+  ]);
+});
+
+test('.getPayloadExamples() should return examples from payload schema', t => {
+  const result = getPayloadExamples(
+    new Message({
+      payload: {
+        examples: [{ foo: 'bar' }, { bar: 'foo' }],
+      },
+    }),
+  );
+  t.deepEqual(result, [
+    {
+      example: { foo: 'bar' },
+    },
+    {
+      example: { bar: 'foo' },
+    },
+  ]);
+});
+
+test('.getPayloadExamples() should return examples from payload schema - case when headers examples are defined in `examples` field', t => {
+  const result = getPayloadExamples(
+    new Message({
+      payload: {
+        examples: [{ foo: 'bar' }, { bar: 'foo' }],
+      },
+      examples: [
```

Comment:
this should be returned, not `payload.examples` as it is explicitly provided for a message

Reply:
Do you mean `name` and `surname`?

Reply:
`surname`? I think it is better to say `last name` 😄 

I mean that `examples` have precedence over `payload.examples`

Reply:
2nd vaccine shot didn't go as planned, huh 😄 

> Where you see payload in examples? 😄

I did not say there is a payload in examples, but examples in payload 😉

Reply:
If we don't have any `payload` in examples, we look for `payload.examples`.

Reply:
omg, now I get it 😄 damn 🤦🏼 I didn't notice payload is not there, same https://github.com/asyncapi/generator-filters/pull/41/files#diff-6eda616dc7059166fa2001f838f2a48451f7e3b708279c8ca7397b8d8eb104f4R201, headers are not there

but then it means we need another 2 tests that pick `examples` if they are correct, even if `payload.examples` is provided

Reply:
Done :)